Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs

```diff
@@ -5560,139 +5699,134 @@ ObservableList<ElementLocator> IWindowExplorer.GetElementFriendlyLocators(Elemen
             IWebElement NearwebElement = null;
             try
             {
-                if (((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling != null && ((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling.Name.StartsWith("#"))
+                if (((HTMLElementInfo)ElementInfo).HTMLElementObject != null)
                 {
-                    ((HTMLElementInfo)ElementInfo).RightofHTMLElementObject = ((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling.NextSibling;
-                    RightofwebElement = ((HTMLElementInfo)ElementInfo).RightofHTMLElementObject != null ? Driver.FindElement(By.XPath(((HTMLElementInfo)ElementInfo).RightofHTMLElementObject.XPath)) : null;
-                    ElementLocator elemLocator = new ElementLocator();
-                    elemLocator.Active = true;
-                    elemLocator.Position = ePosition.RightOf;
-                    elemLocator.FriendlyObject = RightofwebElement;
-                    elemLocator.LocateBy = eLocateBy.POMElement;
-                    elemLocator.LocateValue = ((HTMLElementInfo)ElementInfo).RightofHTMLElementObject != null ? (((HTMLElementInfo)ElementInfo).RightofHTMLElementObject.Name == "div" ? String.Empty : ((HTMLElementInfo)ElementInfo).RightofHTMLElementObject.XPath) : String.Empty;
-                    elemLocator.IsAutoLearned = true;
-                    if (!string.IsNullOrEmpty(elemLocator.LocateValue))
+                    if (((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling != null && ((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling.Name.StartsWith("#"))
                     {
-                        locatorsList.Add(elemLocator);
+                        ((HTMLElementInfo)ElementInfo).LeftofHTMLElementObject = ((HTMLElementInfo)ElementInfo).HTMLElementObject.NextSibling.NextSibling;
+                        RightofwebElement = ((HTMLElementInfo)ElementInfo).LeftofHTMLElementObject != null ? Driver.FindElement(By.XPath(((HTMLElementInfo)ElementInfo).LeftofHTMLElementObject.XPath)) : null;
+                        ElementLocator elemLocator = new ElementLocator();
+                        elemLocator.Active = true;
+                        elemLocator.Position = ePosition.LeftOf;
```

Comment:
see if we can optimize the duplicate code of if and else

Reply:
Done